claro-vim/syntax/claro.vim

```diff
@@ -0,0 +1,62 @@
+" https://learnvimscriptthehardway.stevelosh.com/chapters/44.html
+
+if exists("b:current_syntax")
+    finish
+endif
+
+syntax match claroConstant "\v[a-zA-Z][a-zA-Z0-9]*"
+
+syntax match claroType "\v\["
+syntax match claroType "\v\]"
+syntax match claroType "\v\{"
+syntax match claroType "\v\}"
+syntax keyword claroType NothingType int boolean float future lambda string
+
+syntax keyword claroConditional else if match case or and not return
+syntax keyword claroRepeat for while break continue repeat
+
+syntax match claroNumber "\v[0-9]\.?"
+
+syntax match claroCharacter "\v\'.\'"
+
+syntax match claroOperator "\v\="
+syntax match claroOperator "\v\!\="
+syntax match claroOperator "\v\?\="
+syntax match claroOperator "\v\+\+"
+syntax match claroOperator "\v--"
+syntax match claroOperator "\v,"
+syntax match claroOperator "\v:"
+syntax match claroOperator "\v;"
+syntax match claroOperator "\v\^"
+syntax match claroOperator "\v\*"
+syntax match claroOperator "\v\/"
+syntax match claroOperator "\v\+"
+syntax match claroOperator "\v-"
+syntax match claroOperator "\v-\>"
+syntax match claroOperator "\v\<-"
+syntax match claroOperator "\v\>"
+syntax match claroOperator "\v\<"
+syntax match claroOperator "\v\>\="
+syntax match claroOperator "\v\<\="
+syntax match claroOperator "\v\=\="
+
+syntax match claroIdentifier "\v[a-zA-Z0-9]*"
+
+syntax match claroFunction "\v^[a-z][a-zA-Z0-9]*"
```

Comment:
Claro also supports underscores in identifiers, here's the [regex the lexer uses](https://github.com/JasonSteving99/claro-lang/blob/4987344526b1582ce3d202eeb1ac0a3c98e97791/src/java/com/claro/ClaroLexer.flex#L88). Would you mind adding that?

Reply:
sure thing!